### DISCUSSION TITLE
UI Grid Edit Mode is disabled on multiple union selects

### DISCUSSION THREAD

**Discussion by noldoreldar** [dbeaver/dbeaver#36494] [Q&A]
Database: PostgreSQL 16.04
DBeaver: Community 23.3.1
Driver: PostgreSQL JDBC Driver

On SQL Query Window, I run a select query over one single table with calculated columns. Such as; 

`select t.id, t.name, 'xyz' as "custom" from table1 t where t.id = 123`

On the grid results, editing a column value work just fine in this situation. I can edit "name" column value and save the row from grid. 

However, when I "union" multiple select queries with exact same columns just with different Ids in where condition, the grid fully turns into read-only mode and I can not edit anymore. Such as;

`select t.id, t.name, 'xyz' as "custom" from table1 t where t.id = 123`
`union all`
`select t.id, t.name, 'abc' as "custom" from table1 t where t.id = 234`
`union all`
`select t.id, t.name, 'qwe' as "custom" from table1 t where t.id = 789`

Is there any work around for this situation from settings? Or can you think of something else I could do with my query not to lose the row references and still I can edit?

Thank you.

**Comment by ged-yuko** [comment]
The approach would depend on the particular logic behind "custom values" to associate, so my first approach would have been something like this:
```
select t.id, t.name, (
    case t.id
        when 123 then 'xyz'
        when 234 then 'abc'
        when 789 then 'qwe'
    end
) as "custom" 
from table1 t
where t.id in (
    123, 234, 789
)
```
